<launch>
	<!-- declare arguments with default values -->
	<arg name="proto_file" default="$(find road_detect)/models/mssd_300_deploy.prototxt"/>
	<arg name="model_file" default="$(find road_detect)/models/mssd_300_deploy.caffemodel"/>
	<arg name="model_name" default="mssd_300"/>
    	<arg name="image_source_topic" default="/usb_cam/image_raw"/>
    	<arg name="cam_cmd_topic" default="/cam/cmd"/>
    	<arg name="image_object_topic" default="/road_obj"/>
    	<arg name="road_status_topic" default="/road/status"/>
    	<arg name="road_active_topic" default="/road/active"/>

	<node pkg="road_detect" type="road_detect" name="road_detect" output="screen">
	<param name="proto_file" value="$(arg proto_file)"/>
	<param name="model_file" value="$(arg model_file)"/>
	<param name="model_name" value="$(arg model_name)"/>
        <param name="image_source_topic" value="$(arg image_source_topic)"/>
        <param name="cam_cmd_topic" value="$(arg cam_cmd_topic)"/>
        <param name="image_object_topic" value="$(arg image_object_topic)"/>
        <param name="road_status_topic" value="$(arg road_status_topic)"/>
        <param name="road_active_topic" value="$(arg road_active_topic)"/>
	</node>
</launch>
